security.md

```diff
@@ -1,60 +1,172 @@
+possible.
+
+**Some non-secrets should still be kept out of the public domain.** 
+"Private" information that does not constitute a secret but may make
+an exploit easier to achieve should be kept in private repositories.
+
+Also see the guidance on 
+[github repository contents](https://github.com/guardian/recommendations/blob/main/github.md#repository-contents)
+
+
 
 ## AWS
 
+Centralised Reporting from AWS services and guidance is available in
+[Security HQ](https://security-hq.gutools.co.uk).
+
+### Security patches for Compute services
+See the [patching guidance below](#patching)
+
+### Credentials
+
 **Do not use permanent AWS credentials.** Applications running in AWS
 should use `assumeRole` to gain access to other AWS resources. For
 local development you should use AWS profiles to manage credentials
 and these should be temporary credentials (e.g. provided by Janus).
 
+**Permanent credentials should be rotated every 365 days.** (When they
+cannot be replaced by temporary credentials.)
+
 **Ensure that at least 2 people have IAM logins for each AWS account.**  
 In the event of a VPN or Janus outage this will ensure that teams can
-still access their account. IAM users should have 2FA enabled and no
+still access their account. These IAM users should have 2FA enabled and no
 active access keys.
 
-**Use Security HQ to get an overview of our AWS accounts.**
-Centralised Reporting from AWS services, including Trusted Advisor, is
-available in [Security HQ](https://security-hq.gutools.co.uk) and can be
-used to monitor the security status of our AWS accounts.
+### SSH Access
+
+**Direct access to production infrastructure should be a last resort.** 
+Prefer automating common actions and shipping telemetry data extrernally,
+which should reduce the need.
 
 **Use SSM-Scala for SSH access.**
 [SSM-scala](https://github.com/guardian/ssm-scala#ssm-scala) is a tool
 executing commands on EC2 instances authenticated by IAM credentials.
 The SSM Agent is baked into many of our AMI images and should be
 preferentially used to permanent keys.
 
+### Security Groups
+
+**Security Groups egress/ingress rules should be locked down as much as
+possible.** There's no reason to allow open access to any EC2 instances; 
+they should ideally be locked down to the load balancer and nothing 
+else. 
+
+**Port 22 access should be removed everywhere.** This is no longer required
+for SSH access via SSM-scala (see above.)
+
+### S3 Buckets
+
+It is surprisingly easy to accidentally make public an S3 bucket or object
+public. There are a number of ways of controlling access, all of which must
+be correctly configured to keep private objects private.
+
+**Prefer using a bucket for a single logical purpose, either public or 
+private.** Mixing concerns makes it more complicated to configure access and
+easier to get it wrong. This allows access to be configured at the bucket 
+level, which is preferred to the object level.
```

Comment:
Should we recommend granting public access objects should be done via the bucket's policy? (It's a _lot_ simpler than a policy per object). This also reinforces the idea that a bucket is either entirely private or entirely public.

Reply:
I think that is what I'm trying to recommend with `This allows access to be configured at the bucket level, which is preferred to the object level`.

Does that read ok to you - shall I reword it?

Reply:
Hmm, not sure why I didn't notice that. Maybe we can split this in two?
  1. Prefer single, logical purpose buckets
  2. Control use of bucket policies to grant access